proto/dd_metric.proto

```diff
@@ -40,7 +51,7 @@ message MetricPayload {
         string unit = 6;
         string source_type_name = 7;
         int64 interval = 8;
-        reserved 9;
+        Metadata metadata = 9;
```

Comment:
(addressed in [b4190b2](https://github.com/vectordotdev/vector/pull/18405/commits/b4190b220361ade19162f63498b1e26f718489b9))

Reply:
In this case, this change okay. `9` was previously reserved because we didn't expose this field. Reserving the index prevented us from accidentally assigning it to a different field, which would break compatibility. However, because we are changing it from reserved back to the "correct" type/field, this change is okay. So we'll want to change this back to 9. Otherwise, the `metadata` field would be ignored by the backend.

Reply:
This has me wondering about the check in CI 🤔

Reply:
The exit code was 100

Reply:
It's important for us to have the CI check to flag any possible compatibility-breaking changes, which this is an example.  A majority of the time, the changes flagged are indeed breaking changes. In certain scenarios, the changes may actually be safe but require further inspection to verify that they are compatible. This PR is an example of the latter. The `reserved` keyword is used to denote that an index should not be used. Thus, removing the `reserved 9` statement and assigning it to a type is a potentially breaking change. In this case, it is okay because we know that it is the correct type. However, it would be breaking had we set it to any other type. This is not something that the checker can determine itself. Thus, we have to override the checker, similar to how we might add a lint rule override in code. For context, I've use these types of checks in previous organizations and they have saved us from many accidental breakages 🙂

Reply:
I reverted the commit so it's back to what I had before, replacing the reserved field.